src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -614,6 +614,59 @@ internal static void RegisterMemoryLoadChangeNotification(float lowMemoryPercent
             }
         }
 
+        [Flags]
+        internal enum NoGCRegionCallbackWork
+        {
+            None,
+            Execute,
+            Detach
+        }
+
+        internal enum EnableNoGCRegionCallbackStatus
+        {
+            Succeed,
+            InsufficientBudget
+        }
+
+        private static Action? _NoGCCallback;
+
+        [UnmanagedCallersOnly]
+        private static unsafe void NoGCRegionCallback(NoGCRegionCallbackWork work)
+        {
+            Action? callback = _NoGCCallback;
+            if (work.HasFlag(NoGCRegionCallbackWork.Detach))
```

Comment:
```suggestion
            if ((work & NoGCRegionCallbackWork.Detach) != 0)
```
It is better to use idiomatic C# for bit tests. The JIT has to do extra work to optimize `HasFlag` back into a simple bit test, and this optimization does not always happen.